source/auth/tests/connection-string.yml

```diff
@@ -2,366 +2,336 @@ tests:
     -
         description: "should use the default source and mechanism"
         uri: "mongodb://user:password@localhost"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user"
             password: "password"
-            db: "admin"
-        options: ~
+            source: "admin"
+            mechanism: "default"
+            mechanism_properties: {}
     -
         description: "should use the database when no authSource is specified"
         uri: "mongodb://user:password@localhost/foo"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user"
             password: "password"
-            db: "foo"
-        options: ~
+            source: "foo"
+            mechanism: "default"
+            mechanism_properties: {}
     -
         description: "should use the authSource when specified"
         uri: "mongodb://user:password@localhost/foo?authSource=bar"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user"
             password: "password"
-            db: "bar"
-        options: ~
+            source: "bar"
+            mechanism: "default"
+            mechanism_properties: {}
     -
         description: "should recognise the mechanism (GSSAPI)"
         uri: "mongodb://user%40DOMAIN.COM@localhost/?authMechanism=GSSAPI"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user@DOMAIN.COM"
-            password: ~
-            db: "$external"
-        options:
-            authmechanism: "GSSAPI"
+            password: ""
+            source: "$external"
+            mechanism: "GSSAPI"
+            mechanism_properties:
+                SERVICE_NAME: "mongodb"
```

Comment:
Remove, as this mechanism property is not configured in the connection string.

Reply:
Are you thinking that the connection string should be aware of the default values for mechanism properties?  If so, let's discuss.  If not, please remove from here and below.

Reply:
This is not a test of connection string parsing alone.  It's a test of the auth spec itself, which indicates default values for attributes of a Credential object.  Each test is a series of assertions: given a URI, what auth *Credential object* is constructed.

This is why, for instance, not specifying a database name or authSource in the URI results in a "source" of "admin" up on line 10 (of this diff anyway) – that is the default value per the Auth spec.  This particular test asserts that a Credential has the correct default SERVICE_NAME property if GSSAPI is selected and no SERVICE_NAME is provided.

Reply:
I think this is fine as is

Reply:
Yep, you're right.